lib/llm/http/client.rb

```diff
@@ -4,17 +4,20 @@ module LLM
   module HTTPClient
     def post(req)
       req.content_type = "application/json"
-      response = request(req)
-      response.value
-      response
-    rescue Net::HTTPUnauthorized, Net::HTTPClientException
-      raise LLM::AuthError
-    rescue SystemCallError
-      raise LLM::NetError
-    rescue JSON::ParserError
-      raise LLM::ParseError
-    rescue => e
-      raise LLM::Error, "Unexpected Error: #{e.message}"
+      res = request(req)
+      res.tap(&:value)
+    rescue Net::HTTPClientException
+      if [
+        Net::HTTPBadRequest,   # Gemini (huh?)
+        Net::HTTPForbidden,    # Anthropic
+        Net::HTTPUnauthorized  # OpenAI
+      ].any? { _1 === res }
+        raise LLM::Error::Unauthorized.new { _1.response = res }, "Authentication error"
+      elsif Net::HTTPTooManyRequests === res
+        raise LLM::Error::RateLimit.new { _1.response = res }, "Too many requests"
+      else
+        raise LLM::Error::HTTPError.new { _1.response = res }, "Unexpected response"
```

Comment:
@0x1eef You think it's better to set the message this way? instead of initializing the string in the error class?

Reply:
I'm not sure it is "better" or not :smile:  But usually the caller is in the best place to determine the message to be used. That's where most of the context usually is.